### DISCUSSION TITLE
04: Why do we use softmax and argmax in the train function and only argmax in the test function?

### DISCUSSION THREAD

**Discussion by Lowanski** [mrdbourke/pytorch-deep-learning#916] [Q&A]
Hello everyone!

I hope my question hasn't already been answered. I'm wondering why in part 04 of the course, when we create the train and test functions, we're not consistent in the following steps:
Train fn:
        # Calculate and accumulate accuracy metric across all batches
        y_pred_class = **torch.argmax(torch.softmax(y_pred, dim=1), dim=1)**
        train_acc += (y_pred_class == y).sum().item()/len(y_pred)
Test fn: 
        # Calculate and accumulate accuracy
        test_pred_labels = **test_pred_logits.argmax(dim=1)**
        test_acc += ((test_pred_labels == y).sum().item()/len(test_pred_labels))

I've seen online that softmax doesn't change the result, but why is there a difference here?

**Comment by Lowanski** [comment]
Here is he discussion where it is explained why softmax doesn't change the result : [314](https://github.com/mrdbourke/pytorch-deep-learning/discussions/314)

**Comment by LuluW8071** [comment]
Explained here #875

**Reply by hidenitt4** [reply]
I think the essence of the question was asking why there are two ways of writing what is essentially the same thing. #875 is just a straw-man answer.
